### PR TITLE
Consolidate go modules to top level directory

### LINKED ISSUE #557: Merge CLI and e2e dirs into the root dir
This will prevent for false positives when linting

### PR DESCRIPTION
**Pull request by adam-stokes**
This allows developers who utilize things such as language server protocol to
easily resolve package level signatures across similar modules that make up the
project.

Signed-off-by: Adam Stokes <51892+adam-stokes@users.noreply.github.com>

## What does this PR do?

Moves go modules to the top level directory and each sub directory will reference those imports during build/run phases

## Why is it important?

Gives better developer experience when utilizing linting, language protocols for definition reference, etc

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests for the CLI, and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Author's Checklist

- [ ]

## Related issues

- Closes #557

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,3 +1,26 @@
-module github.com/elastic/e2e-testing
+module github.com/elasticsearch/e2e-testing
```
Comment (mdelapenya) [inline go.mod:1]:
```suggestion
module github.com/elastic/e2e-testing
```

File: go.mod [hunk 2]
```diff
@@ -1,3 +1,26 @@
...
+	github.com/spf13/cobra v1.1.3
+	github.com/stretchr/testify v1.7.0
+	github.com/testcontainers/testcontainers-go v0.10.0
+	go.elastic.co/apm v1.11.0
+	go.elastic.co/apm/module/apmelasticsearch v1.11.0
+	gopkg.in/src-d/go-git.v4 v4.13.1
+	gopkg.in/yaml.v2 v2.4.0
+)
+
+replace github.com/cucumber/godog => github.com/cucumber/godog v0.8.1
```
Comment (mdelapenya) [inline go.mod:26]: I think we need newer versions of godog, at least 0.11 for junit formatters

Comment (adam-stokes) [reply go.mod:26]: Yea waiting on response from @axw in Slack about the import errors, will update once I have this resolved

Comment (adam-stokes) [reply go.mod]: Ok, I believe I fixed the issue, and everything builds as it should

**Comment by adam-stokes** [comment]
Filed this bug to see what if anything could be done with the import errors: https://github.com/elastic/apm-agent-go/issues/922

**Comment by mdelapenya** [comment]
We'd need to consolidate the NOTICE process too

**Comment by adam-stokes** [comment]
@mdelapenya This is now ready for review, go imports fixed as well

**Review by mdelapenya — APPROVED** [review]

**Pull request merged**